platform/pulp/app/viewsets/repository.py

```diff
@@ -150,6 +150,16 @@ def destroy(self, request, pk):
 
         return OperationPostponedResponse([async_result])
 
+    @decorators.detail_route()
+    def publish(self, request, pk):
+        publisher = self.get_object()
+        async_result = tasks.publisher.publish.apply_async_with_reservation(
+            tags.RESOURCE_REPOSITORY_TYPE, publisher.repository.name,
```

Comment:
This will raise a 500 if publish is called on a publisher without a repository assigned. (Importers too.) Lets leave it for now and write a bug.